core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,363 @@
+            "Default value %s is not compatible with the expected fixed type, the fixed type is expected to store " +
+                "exactly %s bytes, which means the default value should be of exactly 2 * %s length hex string",
+            defaultValue, fixedLength, fixedLength);
+        byte[] fixedBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.wrap(fixedBytes);
+      case BINARY:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        byte[] binaryBytes = BaseEncoding.base16().decode(defaultValue.textValue().toUpperCase(Locale.ROOT));
+        return ByteBuffer.wrap(binaryBytes);
+      case LIST:
+        Preconditions.checkArgument(defaultValue.isArray(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        Type elementType = type.asListType().elementType();
+        return Lists.newArrayList(Iterables.transform(defaultValue, e -> fromJson(elementType, e)));
+      case MAP:
+        Preconditions.checkArgument(
+            defaultValue.isObject() && defaultValue.has(KEYS) && defaultValue.has(VALUES) &&
+                defaultValue.get(KEYS).isArray() && defaultValue.get(VALUES).isArray(),
+            "Cannot parse %s to a %s value",
+            defaultValue, type);
+        JsonNode keys = defaultValue.get(KEYS);
+        JsonNode values = defaultValue.get(VALUES);
+        Preconditions.checkArgument(
+            keys.size() == values.size(),
+            "Cannot parse default as a %s value: %s",
+            type,
+            defaultValue);
+
+        ImmutableMap.Builder<Object, Object> mapBuilder = ImmutableMap.builder();
+
+        Iterator<JsonNode> keyIter = keys.iterator();
+        Type keyType = type.asMapType().keyType();
+        Iterator<JsonNode> valueIter = values.iterator();
+        Type valueType = type.asMapType().valueType();
+
+        while (keyIter.hasNext()) {
+          mapBuilder.put(fromJson(keyType, keyIter.next()), fromJson(valueType, valueIter.next()));
+        }
+
+        return mapBuilder.build();
+      case STRUCT:
+        Preconditions.checkArgument(defaultValue.isObject(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        Types.StructType struct = type.asStructType();
+        StructLike defaultRecord = GenericRecord.create(struct);
+
+        List<Types.NestedField> fields = struct.fields();
+        for (int pos = 0; pos < fields.size(); pos += 1) {
+          Types.NestedField field = fields.get(pos);
+          String idString = String.valueOf(field.fieldId());
+          if (defaultValue.has(idString)) {
+            defaultRecord.set(pos, fromJson(field.type(), defaultValue.get(idString)));
+          }
+        }
+        return defaultRecord;
+      default:
+        throw new IllegalArgumentException(String.format("Type: %s is not supported", type));
```

Comment:
This should also be `UnsupportedOperationException`

Reply:
fixed